### PR TITLE
ip_diffim tickets/DM-2535: switch from compound fields to FunctorKeys

### PR DESCRIPTION
**Pull request by TallJimbo**

### CODE REVIEW

File: tests/diaCatalogSourceSelector.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def makeMatches(self, refCat, srcCat, nSrc):
             refSrc.set("r_flux", 10**(-0.4*18))
             refSrc.set("resolved", False)
             refSrc.set("photometric", True)
-            refSrc.set("coord", coord)
+            refSrc.setCoord(coord)
```
Comment (r-owen) [inline tests/diaCatalogSourceSelector.py]: Do you plan to support setCoord for the long haul? If not, I suggest a key.

Comment (TallJimbo) [inline tests/diaCatalogSourceSelector.py]: No, I don't.  I'll change it.

Comment (TallJimbo) [inline tests/diaCatalogSourceSelector.py]: On second thought, I'm going to leave it for now.  I'm actually deferring the removal of those setters and getters for a while, and part of the problem is that I know I need to find a better syntax to replace them, and I don't have that yet.  The only alternative that would work right now in this case is really verbose, which just reemphasizes that problem.

**Comment by TallJimbo** [comment]
Merged to master on tickets/DM-1766.

**Pull request closed without merging**